src/jit/morph.cpp

```diff
@@ -7043,31 +7249,145 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
             }
             else
             {
-                hasMultiByteArgs = true;
+                hasMultiByteStackArgs = true;
             }
         }
+        else
+        {
+            varTypeIsFloating(argx) ? ++calleeFloatArgRegCount : ++calleeArgRegCount;
+        }
+
+#if !DEBUG
+        // We can break early on multiByte cases.
```

Comment:
This is an optimization outside of debug to not continue calculations if we know that we will fail the decision. While in debug I found it useful to accurately count register usage and stack usage.